sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -186,7 +186,7 @@ case class AdaptiveSparkPlanExec(
     // created in the middle of the execution.
     context.session.withActive {
       val executionId = getExecutionId
-      var currentLogicalPlan = currentPhysicalPlan.logicalLink.get
+      var currentLogicalPlan = inputPlan.logicalLink.get
```

Comment:
I think it's a better place to fix this issue

Reply:
Could you also add a simple comment mentioning why we use `inputPlan` instead of `currentPhysicalPlan` ?

Reply:
sure, added